### PR TITLE
Joe/winarm investigation

### PR DESCRIPTION
**Pull request by joe-braley**
Merge the branch for releases into `microsoft-main`. `microsoft-main` was chosen so that `main` could reflect upstream and we can then pull changes into `microsoft-main` as needed.

Thoughts for improvement:
  - Remove the platform specific code and replace with Python.
  - There will be a need for special handling of Windows since that platform relies on Cygwin and the `sh` scripts seem to have troubles on newer versions of Windows.
  - Add naming to the top level job to easily determine what platform is being tested.

### CODE REVIEW

File: .github/workflows/paralleltest-docker.yml [hunk 1]
```diff
@@ -13,14 +13,20 @@ on:
...
       testLists:
         description: Number of testLists to generate
         required: false
-        default: 4
+        default: '4'
+      pool:
+          description: Agent pool
+          required: true
+          type: string
+          default: None
```
Comment (d3r3kk) [inline .github/workflows/paralleltest-docker.yml:28]: Maybe default this to `ubuntu-latest` to leave upstream unaffected?

File: .github/workflows/paralleltest-docker.yml [hunk 2]
```diff
@@ -61,14 +67,27 @@ jobs:
...
+
+      - name: update PATH
+        run: |
+          echo "$ANT_HOME/bin" >> $GITHUB_PATH
+
+      - name: Setup TEST_JDK_HOME
+        working-directory: ${{ github.workspace }}
+        shell: bash
+        run: |
+          curl -s -o linux-binaries.tar.gz "${{ secrets.BASE_CONTAINER_URL }}${{ inputs.customizedSdkUrl }}?${{ secrets.SAS_TOKEN }}"
```
Comment (d3r3kk) [inline .github/workflows/paralleltest-docker.yml:89]: This used to be `${inputs.system}-binaries.tar.gz` - did we remove the `inputs:...system` in an earlier PR?

Comment (d3r3kk) [reply .github/workflows/paralleltest-docker.yml:89]: ...and the linux-binaries.tar.gz that Adoptium produces - would that have included Ant?

File: .github/workflows/paralleltest-docker.yml [hunk 3]
```diff
@@ -13,14 +13,20 @@ on:
...
+        default: '4'
+      pool:
+          description: Agent pool
+          required: true
+          type: string
+          default: None
+        default: '4'
 
 jobs:
   configuration:
```
Comment (d3r3kk) [inline .github/workflows/paralleltest-docker.yml:32]: Should the `runs-on` for the configuration job also be run on the `${ inputs.pool }` set of agents?

File: .github/workflows/paralleltest-docker.yml [hunk 4]
```diff
@@ -61,14 +67,27 @@ jobs:
...
-          curl -s -o ${{ inputs.system }}-binaries.tar.gz "${{ secrets.BASE_CONTAINER_URL }}${{ inputs.customizedSdkUrl }}?${{ secrets.SAS_TOKEN }}"
-          mkdir tmp && tar -xzf ${{ inputs.system }}-binaries.tar.gz -C ./tmp
-          jdk_directory=$(ls ./tmp)
-          echo "TEST_JDK_HOME=${{ github.workspace }}/tmp/$jdk_directory" >> $GITHUB_ENV
+          curl -OLJKSks https://dlcdn.apache.org//ant/binaries/apache-ant-1.9.16-bin.tar.gz
+          curl -o ant-contrib.tar.gz -LJSks https://sourceforge.net/projects/ant-contrib/files/ant-contrib/ant-contrib-1.0b2/ant-contrib-1.0b2-bin.tar.gz/download
+          tar xvf ant-contrib.tar.gz
+          tar xvf apache-ant-1.9.16-bin.tar.gz
+          mv ant-contrib/lib/ant-contrib.jar apache-ant-1.9.16/lib/
+          echo "ANT_HOME=${{ github.workspace }}/apache-ant-1.9.16" >> $GITHUB_ENV
```
Comment (d3r3kk) [inline .github/workflows/paralleltest-docker.yml:79]: Consider putting the download URLs into an ENV var or parameter with a default value. Then, when downloading them into the workspace, consider naming the local file something more generic (or  derive it from the downloaded package URL) so that you won't have to change it whenever the value needs updating.

File: .github/workflows/paralleltest-docker.yml [hunk 5]
```diff
@@ -88,10 +107,12 @@ jobs:
   run-aqa-parallel-docker:
     runs-on: ubuntu-latest
     needs: [setup-parallel-docker, configuration]
+    timeout-minutes: 1440
     container:
-      image: ${{ secrets.CONTAINER_REGISTRY }}
+      image: ${{ inputs.containerRegistry }}
```
Comment (d3r3kk) [inline .github/workflows/paralleltest-docker.yml:112]: Why are we changing this away from the `secrets.CONTAINER_REGISTRY`?

Comment (joe-braley) [reply .github/workflows/paralleltest-docker.yml:112]: This is incorrect. I will revert this change.

File: .github/workflows/paralleltest-docker.yml (hunk 4 above)
Comment (joe-braley) [reply .github/workflows/paralleltest-docker.yml:79]: Will do.

File: .github/workflows/paralleltest-docker.yml (hunk 3 above)
Comment (joe-braley) [reply .github/workflows/paralleltest-docker.yml:32]: `ubuntu-latest` delivers an agent faster. We could use `${ inputs.pool }` to be more consistent and, potentially, more secure.

File: .github/workflows/paralleltest-docker.yml (hunk 2 above)
Comment (joe-braley) [reply .github/workflows/paralleltest-docker.yml:89]: 1) yes, this should be changed to something more generic.

2) No, I do not believe that they package `ant` with their binaries. `ant` is something that needs to be setup. We can investigate leveraging the underlying `run-aqa` js action to install this dependency however.

File: .github/workflows/paralleltest-docker.yml [hunk 6]
```diff
@@ -13,14 +13,20 @@ on:
...
-        default: 'adoptium/aqa-tests:v0.9.7-release'
+        default: 'adoptium/aqa-tests:v0.9.9-release'
       containerRegistry:
         description: Container registry
         required: true
       testLists:
         description: Number of testLists to generate
         required: false
-        default: 4
+        default: '4'
```
Comment (jmjaffe37) [inline .github/workflows/paralleltest-docker.yml:23]: Just curious why the change from `4` to `'4'` was needed. Did this affect compatibility with a script/cli? Why did upstream originally have just `4`?

File: .github/workflows/paralleltest-docker.yml [hunk 7]
```diff
@@ -13,14 +13,20 @@ on:
...
+      pool:
+          description: Agent pool
+          required: true
+          type: string
+          default: None
+        default: '4'
```
Comment (jmjaffe37) [inline .github/workflows/paralleltest-docker.yml:29]: What does this line ( `default: '4'` ) set the default for? It seems to be directly below `default: None`, so I don't see which input it would affect

File: .github/workflows/paralleltest-docker.yml (hunk 3 above)
Comment (jmjaffe37) [reply .github/workflows/paralleltest-docker.yml:32]: I personally vote for `ubuntu-latest` here 😅

File: .github/workflows/paralleltest-docker.yml [hunk 8]
```diff
@@ -109,21 +130,45 @@ jobs:
...
           path: ./
 
-      - name: Setup non-windows environment
+      - name: Setup ANT_HOME
         working-directory: ${{ github.workspace }}
         shell: bash
         run: |
-          curl -s -o ${{ inputs.system }}-binaries.tar.gz "${{ secrets.BASE_CONTAINER_URL }}${{ inputs.customizedSdkUrl }}?${{ secrets.SAS_TOKEN }}"
-          mkdir tmp && tar -xzf ${{ inputs.system }}-binaries.tar.gz -C ./tmp
+          curl -OLJKSks https://dlcdn.apache.org//ant/binaries/apache-ant-1.9.16-bin.tar.gz
```
Comment (jmjaffe37) [inline .github/workflows/paralleltest-docker.yml:137]: This is the same code as above right? Might this be better as a template?

File: .github/workflows/paralleltest.yml [hunk 9]
```diff
@@ -17,6 +17,13 @@ on:
       system:
         description: System being tested
         required: true
+      version:
+        description: JDK version
+        required: false
+        type: choice
+        options:
```
Comment (jmjaffe37) [inline .github/workflows/paralleltest.yml:24]: Should there be an option for `jdk21` now? Or does that require additional development?

File: .github/workflows/paralleltest.yml [hunk 10]
```diff
@@ -69,14 +76,23 @@ jobs:
...
         suite: [functional, openjdk, system, perf]
     steps:
+      - name: clean up workspace (non-windows)
+        working-directory: ${{ github.workspace }}
+        shell: bash
+        if: ${{ inputs.system != 'windows' }}
+        run: |
+          ls -la ./
+          rm -rf ./*
+          rm -rf ./.??*
```
Comment (jmjaffe37) [inline .github/workflows/paralleltest.yml:99]: just curious what `./.??*` will do/find

File: .github/workflows/paralleltest.yml [hunk 11]
```diff
@@ -116,15 +132,26 @@ jobs:
...
+          ls -la ./
```
Comment (jmjaffe37) [inline .github/workflows/paralleltest.yml:158]: The `cleanup workspace` step above omits the `ls -la ./` step at the end. Did you mean to include that there, or omit this one?

**Review by jmjaffe37 — APPROVED** [review]
LGTM, I only have nitpicks and minor questions

File: .github/workflows/paralleltest-docker.yml (hunk 8 above)
Comment (jmjaffe37) [reply .github/workflows/paralleltest-docker.yml:137]: Meaning: lines 137 through 142 compared to 74 through 79

File: .github/workflows/paralleltest-docker.yml (hunk 3 above)
Comment (d3r3kk) [reply .github/workflows/paralleltest-docker.yml:32]: Because we are asked to use Microsoft 1ES / deemed secure agents in everything we do, I would say keep it to `${ inputs.pool }`. 

We can always provide the 1ES team with more hit-counts on any bugs they are monitoring about provisioning time, and it is our responsibility to do this kind of thing rather than working around the intended system anyways :).

**Review by d3r3kk — CHANGES_REQUESTED** [review]
Setting my review to 'request changes' so I'll be notified when to go over this once more ;)

**Comment by joe-braley** [comment]
Going to close this pull request as we will be introducing more changes shortly.
